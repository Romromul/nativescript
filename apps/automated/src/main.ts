let start;
if (typeof NSDate !== 'undefined') {
	start = NSDate.date();
} else {
	start = java.lang.System.currentTimeMillis();
}

import { AndroidActivityBackPressedEventData, AndroidActivityBundleEventData, AndroidActivityEventData, AndroidActivityNewIntentEventData, AndroidActivityResultEventData, Application, ApplicationEventData, DiscardedErrorEventData, LaunchEventData, UnhandledErrorEventData } from '@nativescript/core';

if (Application.ios) {
	// Observe application notifications.
	Application.ios.addNotificationObserver(UIApplicationDidFinishLaunchingNotification, (notification: NSNotification) => {
		console.log('UIApplicationDidFinishLaunchingNotification:', notification);
	});
}

// Common events for both Android and iOS.
Application.on(Application.displayedEvent, function (args: ApplicationEventData) {
	global.isDisplayedEventFired = true;

	if (args.android) {
		// For Android applications, args.activity is an Android activity class.
		console.log('Displayed Activity:', (args as AndroidActivityEventData).activity);
	} else if (args.ios) {
		// For iOS applications, args.ios is UIApplication.
		console.log('Displayed UIApplication:', args.ios);
	}
});

Application.on(Application.launchEvent, function (args: LaunchEventData) {
	if (args.android) {
		// For Android applications, args.android is an android.content.Intent class.
		console.log('Launched Android application with the following intent:', args.android);
	} else if (args.ios !== undefined) {
		// For iOS applications, args.ios is NSDictionary (launchOptions).
		console.log('Launched iOS application with options:', args.ios);
	}
});

Application.on(Application.suspendEvent, function (args: ApplicationEventData) {
	if (args.android) {
		// For Android applications, args.android is an Android activity class.
		console.log('Suspend Activity:', args.android);
	} else if (args.ios) {
		// For iOS applications, args.ios is UIApplication.
		console.log('Suspend UIApplication:', args.ios);
	}
});

Application.on(Application.resumeEvent, function (args: ApplicationEventData) {
	if (args.android) {
		// For Android applications, args.android is an Android activity class.
		console.log('Resume Activity:', args.android);
	} else if (args.ios) {
		// For iOS applications, args.ios is UIApplication.
		console.log('Resume UIApplication:', args.ios);
	}
});

Application.on(Application.exitEvent, function (args: ApplicationEventData) {
	if (args.android) {
		// For Android applications, args.android is an Android activity class.
		console.log('Exit Activity:', args.android);
	} else if (args.ios) {
		// For iOS applications, args.ios is UIApplication.
		console.log('Exit UIApplication:', args.ios);
	}
});

Application.on(Application.lowMemoryEvent, function (args: ApplicationEventData) {
	if (args.android) {
		// For Android applications, args.android is an Android activity class.
		console.log('Low Memory:', args.android);
	} else if (args.ios) {
		// For iOS applications, args.ios is UIApplication.
		console.log('Low Memory:', args.ios);
	}
});

// Error events.
Application.on(Application.uncaughtErrorEvent, function (args: UnhandledErrorEventData) {
	console.log('NativeScriptError:', args.error);
	console.log((<any>args.error).nativeException ?? (<any>args.error).nativeError);
	console.log((<any>args.error).stackTrace ?? (<any>args.error).stack);
});

Application.on(Application.discardedErrorEvent, function (args: DiscardedErrorEventData) {
	console.log('[Discarded] NativeScriptError:', args.error);
	console.log((<any>args.error).nativeException ?? (<any>args.error).nativeError);
	console.log((<any>args.error).stackTrace ?? (<any>args.error).stack);
});

// Android activity events.
if (Application.android) {
	Application.android.on(Application.android.activityCreatedEvent, function (args: AndroidActivityBundleEventData) {
		console.log('Event: ' + args.eventName + ', Activity:', args.activity, ', Bundle:', args.bundle);
	});

	Application.android.on(Application.android.activityDestroyedEvent, function (args: AndroidActivityEventData) {
		console.log('Event: ' + args.eventName + ', Activity: ', args.activity);
	});

	Application.android.on(Application.android.activityStartedEvent, function (args: AndroidActivityEventData) {
		console.log('Event: ' + args.eventName + ', Activity:', args.activity);
	});

	Application.android.on(Application.android.activityPausedEvent, function (args: AndroidActivityEventData) {
		console.log('Event: ' + args.eventName + ', Activity:', args.activity);
	});

	Application.android.on(Application.android.activityResumedEvent, function (args: AndroidActivityEventData) {
		console.log('Event: ' + args.eventName + ', Activity:', args.activity);
	});

	Application.android.on(Application.android.activityStoppedEvent, function (args: AndroidActivityEventData) {
		console.log('Event: ' + args.eventName + ', Activity:', args.activity);
	});

	Application.android.on(Application.android.saveActivityStateEvent, function (args: AndroidActivityBundleEventData) {
		console.log('Event: ' + args.eventName + ', Activity:', args.activity, ', Bundle:', args.bundle);
	});

	Application.android.on(Application.android.activityResultEvent, function (args: AndroidActivityResultEventData) {
		console.log('Event:', args.eventName, ', Activity:', args.activity, ', requestCode: ', args.requestCode, ', resultCode: ', args.resultCode, ', Intent: ', args.intent);
	});

	Application.android.on(Application.android.activityBackPressedEvent, function (args: AndroidActivityBackPressedEventData) {
		console.log('Event:', args.eventName, ', Activity:', args.activity);
		// Set args.cancel = true to cancel back navigation and do something custom.
	});

	Application.android.on(Application.android.activityNewIntentEvent, function (args: AndroidActivityNewIntentEventData) {
		console.log('Event: ', args.eventName, ', Activity:', args.activity, ', Intent:', args.intent);
	});
}

let time;
if (typeof NSDate !== 'undefined') {
	time = NSDate.date().timeIntervalSinceDate(start) * 1000;
} else {
	time = java.lang.System.currentTimeMillis() - start;
}

console.log(`TIME TO LOAD APP: ${time} ms`);

Application.run({ moduleName: 'app-root' });
