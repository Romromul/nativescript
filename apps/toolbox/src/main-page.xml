<Page xmlns="http://schemas.nativescript.org/tns.xsd" navigatingTo="navigatingTo" class="page">
    <Page.actionBar>
        <ActionBar title="Dev Toolbox" icon="" class="action-bar">
        </ActionBar>
    </Page.actionBar>
    <ScrollView>
    <StackLayout class="p-t-20 p-x-20">

        <!-- <Button text="Button with html boxShadow" tap="{{ onTap }}" height="50" class="btn btn-primary btn-active" boxShadow="5 5 10 navy"/>  

        <Button text="Button with css boxShadow (rgba)" tap="{{ onTap }}" height="50" marginTop="50" class="btn btn-primary btn-active"/>  
  
        <Button text="Button with boxShadow 3 props" tap="{{ onTap }}" height="50" marginTop="50" boxShadow="5 5 navy" class="btn btn-primary btn-active"/>  
        <Button text="Button with boxShadow 4 props" tap="{{ onTap }}" height="50" marginTop="50" boxShadow="5 5 10 navy" class="btn btn-primary btn-active"/>  
        <Button text="Button with boxShadow 5 props" tap="{{ onTap }}" height="50" marginTop="50" boxShadow="5 5 10 10 navy" class="btn btn-primary btn-active"/>  

        <StackLayout boxShadow="10 10 rgba(0,0,0,1)" marginTop="50">
          <Button text="Button with css boxShadow" tap="{{ onTap }}" height="50" borderRadius="10"/>
        </StackLayout> -->

        <!-- TODO: if backgroundColor is not set, it won't call background.ios.ts hence not applying the boxShadow -->
        <!-- <StackLayout boxShadow="5 5 10 10 red" height="100" backgroundColor="transparent" padding="10" margin="20">
            <Label text="StackLayout with transparent background"></Label>
        </StackLayout> -->

        <GridLayout boxShadow="10 -10 10 10 rgba(0,0,0,0.5)" height="100" backgroundColor="lightblue" padding="10" margin="20" tap="{{ toggleAnimation }}">
            <Label text="GridLayout"></Label>
        </GridLayout>

        <StackLayout boxShadow="5 10 10 20 #000" height="100" backgroundColor="lightblue" padding="10" margin="20" tap="{{ toggleAnimation }}">
            <Label text="StackLayout"></Label>
        </StackLayout>

        <AbsoluteLayout boxShadow="5 15 10 20 green" height="100" backgroundColor="lightblue" padding="10" margin="20" tap="{{ toggleAnimation }}">
            <Label text="AbsoluteLayout"></Label>
        </AbsoluteLayout>

        <!-- note: the 3rd number in box shadow is currently being ignored, only the 1st, 2nd, and 4th, and color are being used-->
        <FlexboxLayout boxShadow="0 0 10 25 red" height="100" backgroundColor="lightblue" padding="10" margin="20" tap="{{ toggleAnimation }}">
            <Label text="FlexboxLayout"></Label>
        </FlexboxLayout>

        <FlexboxLayout boxShadow="15 10 10 20 #000" height="100" backgroundColor="transparent" padding="10" margin="20" tap="{{ toggleAnimation }}">
            <Label text="FlexboxLayout (transparent background)"></Label>
        </FlexboxLayout>

        <Button marginTop="30" boxShadow="0 0 10 8 #000" backgroundColor="transparent" text="button" padding="20"></Button>
        
    </StackLayout>
    </ScrollView>
</Page>
