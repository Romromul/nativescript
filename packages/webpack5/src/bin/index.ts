#!/bin/env node

import { redBright, green, greenBright } from 'chalk';
import { program } from 'commander';
import dedent from 'ts-dedent';
import webpack from 'webpack';
import path from 'path';
import fs from 'fs';

import { parseEnvFlags } from '../cli/parseEnvFlags';

const defaultConfig = path.resolve(
	__dirname,
	'../stubs/default.config.stub.js'
);
const tag = `[${green('@nativescript/webpack')}]`;

function error(message: string) {
	console.error(`${tag} ${redBright(dedent(message))}`);
}

function info(message: string) {
	console.info(`${tag} ${greenBright(dedent(message))}`);
}

program.enablePositionalOptions();

program
	.command('init')
	.description('Initialize a new webpack.config.js in the current directory.')
	.action(() => {
		const targetPath = path.resolve(process.cwd(), 'webpack.config.js');

		if (fs.existsSync(targetPath)) {
			return error(`File Already Exists: ${targetPath}`);
		}

		fs.copyFileSync(defaultConfig, targetPath);

		info('Initialized config.');
	});

program
	.command('build')
	.description('Build...')
	.option('--env [name]', 'environment options')
	.option('--hmr', 'enable HMR')
	.option('--no-hmr', 'disable HMR')
	.option('--watch', 'watch for changes')
	.allowUnknownOption()
	.action((options, command) => {
		const env = parseEnvFlags(command.args);
		// add --env <val> into the env object
		// for example if we use --env prod
		// we'd have env.env = 'prod'
		if (options.env) {
			env['env'] = options.env;
		}
		// const env = {
		// 	platform: 'ios',
		// 	verbose: true,
		// 	appResourcesPath: 'App_Resources',
		// 	appPath: 'src'
		// }

		const configPath = path.resolve(process.cwd(), 'webpack.config.js');
		// todo: validate config exists
		// todo: guard against invalid config
		let configuration;
		try {
			configuration = require(configPath)(env);
		} catch (ignore) {
			console.log(ignore);
		}

		if (!configuration) {
			console.log('No configuration!!!!!');
			return;
		}

		const compiler = webpack(configuration);

		// todo: handle --watch flag
		// todo: promisify callback?
		compiler.watch(
			{
				ignored: ['platforms'],
			},
			(err, stats) => {
				if (stats) {
					console.log(
						stats.toString({
							colors: true,
						})
					);
				}
				// err && console.log(err)
			}
		);
	});

program.parse(process.argv);
