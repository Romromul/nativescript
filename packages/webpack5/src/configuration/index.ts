export { default as angularConfig } from './angular';
export { default as baseConfig } from './base';
export { default as javascriptConfig } from './javascript';
export { default as reactConfig } from './react';
export { default as svelteConfig } from './svelte';
export { default as typescriptConfig } from './typescript';
export { default as vueConfig } from './vue';
