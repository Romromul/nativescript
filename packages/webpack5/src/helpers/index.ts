import { getValue } from './config';
import { getAllDependencies, getDependencyPath } from './dependencies';
import { determineProjectFlavor } from './flavor';
import { error, info, warn } from './log';
import {
	getAbsoluteDistPath,
	getDistPath,
	getEntryPath,
	getPackageJson,
	getPlatform,
	getProjectRootPath,
} from './project';

// intentionally populated manually
// as this generates nicer typings
// that show all the utils inline
// rather than imports to types
export default {
	config: {
		getValue,
	},
	dependencies: {
		getAllDependencies,
		getDependencyPath,
	},
	flavor: {
		determineProjectFlavor,
	},
	log: {
		info,
		warn,
		error,
	},
	project: {
		getProjectRootPath,
		getAbsoluteDistPath,
		getEntryPath,
		getDistPath,
		getPlatform,
		getPackageJson,
	},
};
