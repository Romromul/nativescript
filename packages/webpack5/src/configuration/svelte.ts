import Config from 'webpack-chain';

import { getProjectFilePath } from '../helpers/project';
import { getPlatformName } from '../helpers/platform';
import { env as _env, IWebpackEnv } from '../index';
import { error } from '../helpers/log';
import base from './base';

export default function (config: Config, env: IWebpackEnv = _env): Config {
	base(config, env);

	const platform = getPlatformName();
	const mode = env.production ? 'production' : 'development';
	const production = mode === 'production';

	// target('node') is the default but causes svelte-loader to detect it as a "server" render, disabling HMR
	// electron-main sneaks us past the target == 'node' check and gets us HMR
	config.target('electron-main');

	// svelte-hmr still references tns-core-modules, so we shim it here for compat.
	config.resolve.alias.set('tns-core-modules', '@nativescript/core');

	// resolve .svelte files
	// the order is reversed because we are using prepend!
	config.resolve.extensions.prepend('.svelte').prepend(`.${platform}.svelte`);

	// add worker support to .svelte files (new Worker('./path'))
	config.module.rule('workers').test(/\.(js|ts|svelte)$/);

	// add a rule for .svelte files
	config.module
		.rule('svelte')
		.test(/\.svelte$/)
		.exclude.add(/node_modules/)
		.end()
		.use('svelte-loader')
		.loader('svelte-loader')
		.tap((options) => {
			const svelteConfig = getSvelteConfig();
			return {
				...options,
				compilerOptions: {
					...svelteConfig?.compilerOptions,
					dev: !production,
				},
				preprocess: svelteConfig?.preprocess,
				hotReload: !production,
				hotOptions: {
					injectCss: false,
					native: true,
				},
			};
		});

	return config;
}

interface ISvelteConfig {
	preprocess: any;
	compilerOptions: any;
}

function getSvelteConfig(): ISvelteConfig | undefined {
	try {
		return require(getProjectFilePath('svelte.config.js')) as ISvelteConfig;
	} catch (err) {
		error('Could not find svelte.config.js.', err);
	}
}
