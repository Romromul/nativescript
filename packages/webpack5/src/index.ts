import Config from 'webpack-chain';
import webpack from 'webpack';
import { highlight } from 'cli-highlight';
import { configs } from './configuration';
import { determineProjectFlavor } from './helpers/flavor';
import { applyExternalConfigs } from './helpers/externalConfigs';

export type Platform = 'android' | 'ios' | string;

export interface IWebpackEnv {
	[name: string]: any;

	appPath?: string;
	appResourcesPath?: string;

	android?: boolean;
	ios?: boolean;

	production?: boolean;
	report?: boolean;
	hmr?: boolean;

	// enable verbose output
	verbose?: boolean;
	// todo: add others
}

let webpackChains: any[] = [];
let webpackMerges: any[] = [];
let explicitUseConfig = false;

/**
 * @internal
 */
export let env: IWebpackEnv = {};

////// PUBLIC API
export const defaultConfigs = configs;

export function init(_env: IWebpackEnv) {
	if (_env) {
		env = _env;
	}
}

export function useConfig(config: keyof typeof defaultConfigs | false) {
	explicitUseConfig = true;
	if (config) {
		webpackChains.unshift(configs[config]);
	}
}

export function chainWebpack(
	chainFn: (config: Config, env: IWebpackEnv) => any
) {
	webpackChains.push(chainFn);
}

export function mergeWebpack(
	mergeFn: (
		config: Partial<webpack.Configuration>,
		env: IWebpackEnv
	) => any | Partial<webpack.Configuration>
) {
	webpackMerges.push(mergeFn);
}

export function resolveChainableConfig() {
	const config = new Config();

	if (!explicitUseConfig) {
		useConfig(determineProjectFlavor());
	}

	// apply configs from dependencies
	// todo: allow opt-out
	applyExternalConfigs();

	// this applies all chain configs
	webpackChains.forEach((chainFn) => {
		return chainFn(config, env);
	});

	if (env.verbose) {
		console.log('Resolved chainable config:');
		console.log(highlight(config.toString(), { language: 'js' }));
	}

	return config;
}

export function resolveConfig(chainableConfig = resolveChainableConfig()) {
	// todo: warn if no base config

	// todo: apply merges from webpackMerges

	// return a config usable by webpack
	return chainableConfig.toConfig();
}
