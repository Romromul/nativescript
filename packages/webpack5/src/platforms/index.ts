import { error } from "../helpers/log";
import { env, Platform } from "../";

import AndroidPlatform from "./android";
import iOSPlatform from "./ios";

export interface INativeScriptPlatform {
	getEntryPath?(): string;
	getDistPath?(): string
}

const platforms = {
	android: AndroidPlatform,
	ios: iOSPlatform,
}

export function addPlatform(name: string, platform: INativeScriptPlatform) {
	console.log('adding platform', name, platform)
	platforms[name] = platform;
}

export function getPlatform(): INativeScriptPlatform {
	return platforms[getPlatformName()]
}

export function getPlatformName(): Platform {
	if (env?.android) {
		return 'android';
	}

	if (env?.ios) {
		return 'ios';
	}

	// support custom platforms
	if(env?.platform) {
		return env.platform;
	}

	throw error(`
		You need to provide a target platform!

		Available platforms: ${Object.keys(platforms).join(', ')}

		Use --env=platform=<platform> or --env=android, --env=ios to specify the target platform.
	`);
}
