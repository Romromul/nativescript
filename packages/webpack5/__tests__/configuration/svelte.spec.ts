// @ts-ignore
import Config from 'webpack-chain';
import svelte from '../../src/configuration/svelte';
import { init } from '../../src';

describe.only('svelte configuration', () => {
	const platforms = ['ios', 'android'];

	for (let platform of platforms) {
		it(`for ${platform}`, () => {
			init({
				[platform]: true,
			});
			expect(svelte(new Config()).toString()).toMatchSnapshot();
		});
	}
});
