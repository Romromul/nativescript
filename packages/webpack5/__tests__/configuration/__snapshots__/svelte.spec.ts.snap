// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svelte configuration for android 1`] = `
"{
  mode: 'development',
  externals: [
    'package.json',
    '~/package.json'
  ],
  devtool: 'inline-source-map',
  target: 'node',
  output: {
    path: '__jest__/platforms/android/app/src/main/assets/app',
    pathinfo: false,
    publicPath: '',
    libraryTarget: 'commonjs',
    globalObject: 'global'
  },
  resolve: {
    symlinks: true,
    alias: {
      '~': '__jest__/src',
      '@': '__jest__/src'
    },
    extensions: [
      '.android.svelte',
      '.svelte',
      '.android.ts',
      '.ts',
      '.android.js',
      '.js',
      '.android.css',
      '.css',
      '.android.scss',
      '.scss',
      '.android.json',
      '.json'
    ]
  },
  resolveLoader: {
    modules: [
      'node_modules/@nativescript/webpack/dist/loaders',
      'node_modules'
    ]
  },
  module: {
    rules: [
      /* config.module.rule('ts') */
      {
        test: [
          /\\\\.ts$/
        ],
        use: [
          /* config.module.rule('ts').use('ts-loader') */
          {
            loader: 'ts-loader',
            options: {
              transpileOnly: true,
              allowTsInNodeModules: true,
              compilerOptions: {
                sourceMap: true,
                declaration: false
              },
              getCustomTransformers: function () { /* omitted long function */ }
            }
          }
        ]
      },
      /* config.module.rule('js') */
      {
        test: /\\\\.js$/,
        exclude: [
          /node_modules/
        ],
        use: [
          /* config.module.rule('js').use('babel-loader') */
          {
            loader: 'babel-loader',
            options: {
              generatorOpts: {
                compact: false
              }
            }
          }
        ]
      },
      /* config.module.rule('css') */
      {
        test: /\\\\.css$/,
        use: [
          /* config.module.rule('css').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('css').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          },
          /* config.module.rule('css').use('postcss-loader') */
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: [
                  'postcss-import'
                ]
              }
            }
          }
        ]
      },
      /* config.module.rule('scss') */
      {
        test: /\\\\.scss$/,
        use: [
          /* config.module.rule('scss').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('scss').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          },
          /* config.module.rule('scss').use('postcss-loader') */
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: [
                  'postcss-import'
                ]
              }
            }
          },
          /* config.module.rule('scss').use('sass-loader') */
          {
            loader: 'sass-loader'
          }
        ]
      },
      /* config.module.rule('svelte') */
      {
        test: /\\\\.svelte$/,
        exclude: [
          /node_modules/
        ],
        use: [
          /* config.module.rule('svelte').use('svelte-loader-hot') */
          {
            loader: 'svelte-loader-hot',
            options: {
              dev: true,
              preprocess: undefined,
              hotReload: true,
              hotOptions: {
                injectCss: false,
                'native': true
              },
              onwarn: function () { /* omitted long function */ }
            }
          }
        ]
      }
    ]
  },
  optimization: {
    splitChunks: {
      cacheGroups: {
        defaultVendor: {
          test: /[\\\\\\\\/]node_modules[\\\\\\\\/]/,
          priority: -10,
          name: 'vendor',
          chunks: 'all'
        }
      }
    },
    minimizer: [
      /* config.optimization.minimizer('TerserPlugin') */
      new TerserPlugin(
        {
          terserOptions: {
            compress: {
              collapse_vars: false,
              sequences: false
            },
            keep_fnames: true
          }
        }
      )
    ]
  },
  plugins: [
    /* config.plugin('ForkTsCheckerWebpackPlugin') */
    new ForkTsCheckerWebpackPlugin(
      {
        typescript: {
          memoryLimit: 4096
        }
      }
    ),
    /* config.plugin('CleanWebpackPlugin') */
    new CleanWebpackPlugin(
      {
        cleanOnceBeforeBuildPatterns: [
          '__jest__/platforms/android/app/src/main/assets/app/**/*'
        ],
        verbose: false
      }
    ),
    /* config.plugin('PlatformSuffixPlugin') */
    new PlatformSuffixPlugin(
      {
        platform: 'android'
      }
    ),
    /* config.plugin('FilterWarningsPlugin') */
    new FilterWarningsPlugin(
      {
        exclude: /System.import\\\\(\\\\) is deprecated/
      }
    ),
    /* config.plugin('DefinePlugin') */
    new DefinePlugin(
      {
        __DEV__: true,
        __NS_WEBPACK__: true,
        __CSS_PARSER__: '\\"css-tree\\"',
        __ANDROID__: true,
        __IOS__: false,
        'global.isAndroid': true,
        'global.isIOS': false,
        process: 'global.process'
      }
    ),
    /* config.plugin('CopyWebpackPlugin') */
    new CopyPlugin(
      {
        patterns: [
          {
            from: 'assets/**',
            context: '__jest__/src',
            noErrorOnMissing: true,
            globOptions: {
              dot: false,
              ignore: []
            }
          },
          {
            from: 'fonts/**',
            context: '__jest__/src',
            noErrorOnMissing: true,
            globOptions: {
              dot: false,
              ignore: []
            }
          },
          {
            from: '**/*.+(jpg|png)',
            context: '__jest__/src',
            noErrorOnMissing: true,
            globOptions: {
              dot: false,
              ignore: []
            }
          }
        ]
      }
    ),
    /* config.plugin('WatchStatePlugin') */
    new WatchStatePlugin()
  ],
  entry: {
    bundle: [
      '@nativescript/core/globals/index.js',
      '__jest__/src/app.js'
    ]
  }
}"
`;

exports[`svelte configuration for ios 1`] = `
"{
  mode: 'development',
  externals: [
    'package.json',
    '~/package.json'
  ],
  devtool: 'inline-source-map',
  target: 'node',
  output: {
    path: '__jest__/platforms/ios/jest/app',
    pathinfo: false,
    publicPath: '',
    libraryTarget: 'commonjs',
    globalObject: 'global'
  },
  resolve: {
    symlinks: true,
    alias: {
      '~': '__jest__/src',
      '@': '__jest__/src'
    },
    extensions: [
      '.ios.svelte',
      '.svelte',
      '.ios.ts',
      '.ts',
      '.ios.js',
      '.js',
      '.ios.css',
      '.css',
      '.ios.scss',
      '.scss',
      '.ios.json',
      '.json'
    ]
  },
  resolveLoader: {
    modules: [
      'node_modules/@nativescript/webpack/dist/loaders',
      'node_modules'
    ]
  },
  module: {
    rules: [
      /* config.module.rule('ts') */
      {
        test: [
          /\\\\.ts$/
        ],
        use: [
          /* config.module.rule('ts').use('ts-loader') */
          {
            loader: 'ts-loader',
            options: {
              transpileOnly: true,
              allowTsInNodeModules: true,
              compilerOptions: {
                sourceMap: true,
                declaration: false
              },
              getCustomTransformers: function () { /* omitted long function */ }
            }
          }
        ]
      },
      /* config.module.rule('js') */
      {
        test: /\\\\.js$/,
        exclude: [
          /node_modules/
        ],
        use: [
          /* config.module.rule('js').use('babel-loader') */
          {
            loader: 'babel-loader',
            options: {
              generatorOpts: {
                compact: false
              }
            }
          }
        ]
      },
      /* config.module.rule('css') */
      {
        test: /\\\\.css$/,
        use: [
          /* config.module.rule('css').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('css').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          },
          /* config.module.rule('css').use('postcss-loader') */
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: [
                  'postcss-import'
                ]
              }
            }
          }
        ]
      },
      /* config.module.rule('scss') */
      {
        test: /\\\\.scss$/,
        use: [
          /* config.module.rule('scss').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('scss').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          },
          /* config.module.rule('scss').use('postcss-loader') */
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: [
                  'postcss-import'
                ]
              }
            }
          },
          /* config.module.rule('scss').use('sass-loader') */
          {
            loader: 'sass-loader'
          }
        ]
      },
      /* config.module.rule('svelte') */
      {
        test: /\\\\.svelte$/,
        exclude: [
          /node_modules/
        ],
        use: [
          /* config.module.rule('svelte').use('svelte-loader-hot') */
          {
            loader: 'svelte-loader-hot',
            options: {
              dev: true,
              preprocess: undefined,
              hotReload: true,
              hotOptions: {
                injectCss: false,
                'native': true
              },
              onwarn: function () { /* omitted long function */ }
            }
          }
        ]
      }
    ]
  },
  optimization: {
    splitChunks: {
      cacheGroups: {
        defaultVendor: {
          test: /[\\\\\\\\/]node_modules[\\\\\\\\/]/,
          priority: -10,
          name: 'vendor',
          chunks: 'all'
        }
      }
    },
    minimizer: [
      /* config.optimization.minimizer('TerserPlugin') */
      new TerserPlugin(
        {
          terserOptions: {
            compress: {
              collapse_vars: true,
              sequences: true
            },
            keep_fnames: true
          }
        }
      )
    ]
  },
  plugins: [
    /* config.plugin('ForkTsCheckerWebpackPlugin') */
    new ForkTsCheckerWebpackPlugin(
      {
        typescript: {
          memoryLimit: 4096
        }
      }
    ),
    /* config.plugin('CleanWebpackPlugin') */
    new CleanWebpackPlugin(
      {
        cleanOnceBeforeBuildPatterns: [
          '__jest__/platforms/ios/jest/app/**/*'
        ],
        verbose: false
      }
    ),
    /* config.plugin('PlatformSuffixPlugin') */
    new PlatformSuffixPlugin(
      {
        platform: 'ios'
      }
    ),
    /* config.plugin('FilterWarningsPlugin') */
    new FilterWarningsPlugin(
      {
        exclude: /System.import\\\\(\\\\) is deprecated/
      }
    ),
    /* config.plugin('DefinePlugin') */
    new DefinePlugin(
      {
        __DEV__: true,
        __NS_WEBPACK__: true,
        __CSS_PARSER__: '\\"css-tree\\"',
        __ANDROID__: false,
        __IOS__: true,
        'global.isAndroid': false,
        'global.isIOS': true,
        process: 'global.process'
      }
    ),
    /* config.plugin('CopyWebpackPlugin') */
    new CopyPlugin(
      {
        patterns: [
          {
            from: 'assets/**',
            context: '__jest__/src',
            noErrorOnMissing: true,
            globOptions: {
              dot: false,
              ignore: []
            }
          },
          {
            from: 'fonts/**',
            context: '__jest__/src',
            noErrorOnMissing: true,
            globOptions: {
              dot: false,
              ignore: []
            }
          },
          {
            from: '**/*.+(jpg|png)',
            context: '__jest__/src',
            noErrorOnMissing: true,
            globOptions: {
              dot: false,
              ignore: []
            }
          }
        ]
      }
    ),
    /* config.plugin('WatchStatePlugin') */
    new WatchStatePlugin()
  ],
  entry: {
    bundle: [
      '@nativescript/core/globals/index.js',
      '__jest__/src/app.js'
    ],
    'tns_modules/@nativescript/core/inspector_modules': [
      '@nativescript/core/inspector_modules'
    ]
  }
}"
`;
